curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/65688b5a1655a7a6caede847.md

```diff
@@ -0,0 +1,50 @@
+---
+id: 65688b5a1655a7a6caede847
+title: Step 28
+challengeType: 20
+dashedName: step-28
+---
+
+# --description--
+
+If you try to change the individual characters of a string, you will get a `TypeError`. However, a variable can be reassigned another string:
+
+```py
+message = 'Hello World'
+message = 'Hello there!'
+```
+
+Delete the `text[0]` line and reassign `text` the string `Albatross`.
+
+# --hints--
+
+You should delete `text[0] = 'C'`.
+
+```js
+({ test: () => assert.isFalse( /^text\s*\[\s*0\s*\]\s*\=\s*("|').\1/m.test(e.code.original["main.py"])) })
+```
+
+You should reassign `text` the string `Albatross`. Do not modify `text = 'Hello World'`.
+
+```js
+({ test: () => assert.match(e.code.original["main.py"], /(?<=^text\s*=\s*("|')Hello\sWorld\1.+)^text\s*=\s*("|')Albatross\2/ms) })
```

Comment:
Much to the dev-worlds dismay, only the latest versions of Safari can handle lookbehinds.

So, if a Camper has an older version of Safari, and load this challenge, their page will crash. _Ask me how I know_

```suggestion
({ test: () => assert.match(e.code.original["main.py"].replace(/\r/g, ''), /text\s*=\s*("|')Hello World\1\s*\n^text\s*=\s*("|')Albatross\2/m) })
```

Reply:
I won't ask.